src/hotspot/cpu/aarch64/vm_version_aarch64.cpp

```diff
@@ -574,6 +563,25 @@ void VM_Version::initialize() {
   _spin_wait = get_spin_wait_desc();
 
   check_virtualizations();
+
+  // Sync SVE related CPU features with flags
+  if (UseSVE < 2) {
+    _features &= ~CPU_SVE2;
+    _features &= ~CPU_SVEBITPERM;
+  }
+  if (UseSVE < 1) {
+    _features &= ~CPU_SVE;
+  }
+
+  // Construct the "features" string
+  char buf[512];
+  int buf_used_len = os::snprintf_checked(buf, sizeof(buf), "0x%02x:0x%x:0x%03x:%d", _cpu, _variant, _model, _revision);
+  if (_model2) os::snprintf_checked(buf + buf_used_len, sizeof(buf) - buf_used_len, "(0x%03x)", _model2);
+#define ADD_FEATURE_IF_SUPPORTED(id, name, bit) if (VM_Version::supports_##name()) strcat(buf, ", " #name);
```

Comment:
```suggestion                                                   \
#define ADD_FEATURE_IF_SUPPORTED(id, name, bit)                 \
  do {                                                          \
    if (VM_Version::supports_##name())  strcat(buf, ", " #name); \
  while(0);
```
Reason: no dangling else in macros, ever.

Reply:
Done